### PR TITLE
JS parser improvements

### PR DESCRIPTION
**Pull request by maxfisher-g**
This PR addresses two problems with the JS parser:
1. It could not handle certain variable declarations such as `let [a, b] = [1, 2];`
2. There was no way to differentiate between internal parser errors and syntax errors in the input to be parsed

For 1. the AST traversal was rewritten in the way anticipated in the comment to `traverseAst()` in the existing version of `babel-parser.js`:

> [A] simpler way to do it would be to just traverse until an actual Identifier node is found,
      then figure what kind of identifier it is by inverting the logic for each case that currently is here.

Additional unit tests have been added to `js_parser_test.go` to cover the additional syntax parsing cases.

For 2. when the parser encounters a syntax error in the JS input, it now exits with a pre-defined exit code (33), and produces no output on stdout or stderr. This exit code is checked for in `js_parser.go`, which allows it to set a flag signifying that the input was not valid JavaScript.

### CODE REVIEW

File: internal/staticanalysis/parsing/js/babel-parser.js [hunk 1]
```diff
@@ -25,150 +26,165 @@ function logJSON(type, subtype, name, pos, array, extra = null) {
...
-    for (let p of node.params) {
-        logParameter(p);
+function visitIdentifierOrPrivateName(path) {
+    const node = path.node;
+    const parentNode = path.parentPath.node;
+    const parentParentNode = path.parentPath.parentPath.node;
+
+    switch (parentNode.type) {
+        case "ObjectProperty":
+        // fall through
```
Comment (oliverchang) [inline internal/staticanalysis/parsing/js/babel-parser.js:69]: nit: comment not relevant ?

(also indent is wrong).

File: internal/staticanalysis/parsing/js/babel-parser.js [hunk 2]
```diff
@@ -182,19 +198,32 @@ function findLiteralsAndIdentifiers(source, printDebug) {
 }
 
 function main() {
-    const printDebug = false;
+    const syntaxErrorExitCode = 33;
```
Comment (oliverchang) [inline internal/staticanalysis/parsing/js/babel-parser.js:201]: nit: use SYNTAX_ERROR_EXIT_CODE per https://google.github.io/styleguide/jsguide.html#naming-constant-names

**Review by oliverchang — APPROVED** [review]

File: internal/staticanalysis/parsing/js/babel-parser.js (hunk 1 above)
Comment (maxfisher-g) [reply internal/staticanalysis/parsing/js/babel-parser.js:69]: Will fix in another PR

File: internal/staticanalysis/parsing/js/babel-parser.js (hunk 2 above)
Comment (maxfisher-g) [reply internal/staticanalysis/parsing/js/babel-parser.js:201]: Will fix in another PR

**Pull request merged**